### DISCUSSION TITLE
People profiles not showing up in Search results web part

### DISCUSSION THREAD

**Discussion by Sahitya-trundl** [microsoft-search/pnp-modern-search#4163] [Q&A] [answered]
**Version used**
4.14.0

**Describe the bug**
We are working on implementing a people directory in the Employee Intranet using PnP Modern Search. However, the Search Results web part does not display any people's results. Interestingly, when we use the Search Filters web part to refine the results, the filter result count updates correctly, but the people cards or profiles still do not appear in the results.

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Add PnP - Search Box, PnP - Search Filters and PnP - Search Results to the page
2. Edit "PnP - Search results" web part properties, Select the following in the edit properties pages.
Page: 1 
-- Available Data Sources: Sharepoint Search
-- Result Source Id / Scope|Name: LocalPeopleResults
-- Selected properties: Add **Department** (by default not included)
Page: 2
-- Available Layouts: People
-- Show results count: On
-- Show persona card on hover: On
Page: 3 Under Available connections, 
Use input query text: On
-- Select Dynamic Value
-- Connect to source: PnP Search Box
-- PnP - Search Box's properties: Search Query
-- Check Use a default value
-- Default value: *
Connect to a filters Web Part: On
-- Use filters from this component: Search Filters web part
3. Edit "PnP - Search Filters" web part
Under Available connections, 
-- Use data from these Web Parts: Search results web part
Under Filter settings, Customise filters, Click on Edit
-- Display name: Department
-- Filter field: Department
-- Template: Check Box
-- Multi values: Check
-- Operator between values: OR
Click on Save

Reference document: 
https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/

Youtube reference:
https://youtu.be/VFKs_pISfxg?si=_Nkb3BA5Jz8zv6JR

**Expected behavior**
All the Employee profiles should show up as people cards. Search results should get updated based on the filter selected or the Name added in the Search box

**Screenshots**
Current output without any filter added.
![Screenshot 2024-12-13 152229](https://github.com/user-attachments/assets/cb758f9d-2c31-40b5-93d5-1e04db3f626e)

After the filter is applied, the Search result count gets updated but no people profiles 
![Screenshot 2024-12-13 152430](https://github.com/user-attachments/assets/d81cb0f0-1eb5-415d-a26f-641c7255cbf2)

**Desktop (please complete the following information):**
 - Chrome

**Additional context**

1. No permission issue was detected. 
2. We also tried re-installing the sppkg package but still no luck.
Installation guide: https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.14.0

**Comment by wobba** [comment]
Can you try the tutorial at https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/? And then iterate from that?

**Reply by Sahitya-trundl** [reply]
I followed the same tutorial.

![image](https://github.com/user-attachments/assets/db6b07ca-f689-41cb-9068-265cf841cfcf)

![image](https://github.com/user-attachments/assets/645018f0-5728-4eeb-b2a6-0f085d88b397)

![image](https://github.com/user-attachments/assets/1c8523e5-da0f-48e5-bc5e-fe633cbe23d1)

![image](https://github.com/user-attachments/assets/59eeaa22-f54d-40f8-941d-2c8030ee219f)

![image](https://github.com/user-attachments/assets/7259f647-cf2c-4a47-9a10-c1ec0437089e)

However, the result differs from the expected outcome shown in the tutorial.

Current result:
![image](https://github.com/user-attachments/assets/689447d3-d929-4753-9d2e-7ae75ed6f80a)

**Reply by wobba** [reply]
From your screeshot you have a space after the `:` in the KQL. That breaks parsing. Spec is `<property>:<value>` with no spaces.

**Reply by Sahitya-trundl** [reply]
![image](https://github.com/user-attachments/assets/dff8cdc0-f6bb-4130-a0c2-e130f1eccca0)

I have updated the query but still no luck.

**Reply by wobba** [reply]
Which content source are you using? Works fine when I test it.

<img width="302" alt="image" src="https://github.com/user-attachments/assets/c83fad27-d7b6-400f-982c-5cdeaa18568e" />

**Reply by Sahitya-trundl** [reply]
I am using the same source
![image](https://github.com/user-attachments/assets/60853312-b13d-4172-967d-e2ff552bfdd3)

**Comment by ERTVZ** [comment]
I am getting the same results as @Sahitya-trundl. I am using 4.13.1.

Results count is shown, but persona cards are not. When I take a look in the DevTools I can see that pnp-persona under mgt-person > template > #document-fragment is filled with various attributes and lots of those attributes have data from the corresponding search result. In other words, the pnp-persona shows the data that should be visible in the form of a person card. That means the search results come through in the list items. Each pnp-persona has no childs however. I am not sure if it should have any though.

**Comment by kasperbolarsen** [comment]
Can you test what happens if you use the LPC rather than persona?
![image](https://github.com/user-attachments/assets/323a9009-b468-427c-a672-0e768eeb3504)

**Comment by ERTVZ** [comment]
When I switched "Show persona card on hover (LPC)" _on_ the results started to show. When I switched "Show persone card on hover" _off_ the results showed too (so in this case all "show" toggles where _off_). Switching toggle "Show presence" _on_ works as well. 

So it looks like setting toggle "Show persona card on hover" to _on_ is what causes the display problem.

**Reply by kasperbolarsen** [reply]
It sounds like a permissions issue,  is there anything in the Dev Tools , network,  response?

**Reply by kasperbolarsen** [reply, marked as answer]
Please check if this is enabled
![image](https://github.com/user-attachments/assets/8d8d14b7-71b7-42c8-be27-e5fc0618ffb6)

**Reply by ERTVZ** [reply]
Hi @kasperbolarsen, I couldn't find anything related in the DevTools, At least, if I checked correctly.

When I switched the "Use Microsoft Graph Toolkit" toggle to _on_ (together with "Show persona card on hover") the results started to show. Is that how it's supposed to work? Because the documentation shows the graph toolkit toggle as off ([source](https://microsoft-search.github.io/pnp-modern-search/scenarios/Create-a-useful-People-Search/#layout-configuration)).

**Reply by kasperbolarsen** [reply]
In that case I think that the documention is wrong.
Is there a reason why you prefer "show persona card on hover" over "show persona card on hover (LPC)"? Since "show persona card on hover (LPC)" shows the Out Of The Box people card, I would expect it to be preferred in 95% of the cases, but perhaps I am wrong?
![image](https://github.com/user-attachments/assets/d311e3d7-4cd5-41ad-b55f-bc6495fe8322)

**Reply by Sahitya-trundl** [reply]
It worked when I switched "Show persona card on hover (LPC)" to **ON** the results started to show.
![image](https://github.com/user-attachments/assets/fd3765af-87ee-42ca-90ef-ff6f02dfac20)

**Reply by ERTVZ** [reply]
For me, there's no reason to choose the one above the other. I just happened to stumble upon this discussion and figured I could add my 2 cents. Good to know it works as intended and only the docs are wrong.

**Reply by maober56** [reply]
Hi, I used the tutorial to create a people directory and worked fine, but I have a question, can I use the Microsoft search instead of SharePoint search to accomplish the same, because when I did it, I cannot get any information for people entity but for any others, such Sites or Teams messages. Thank you in advance

**Reply by kasperbolarsen** [reply]
Please see the Q&A section, https://microsoft-search.github.io/pnp-modern-search/QnA/

But no, Microsoft search isn't really able to be used for that purpose
